src/aggregate.d

```diff
@@ -270,35 +276,58 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
As said on #6532, this looks incorrect (or at least surprising/confusing), b/c not all fields were determined.
Just looking at the typical use of the function.
```d
if (!determineFields())
    // error
// reader can expect all fields to be populated, but w/ this early return they are not
```